docs/caching.md

```diff
@@ -0,0 +1,51 @@
+# HTTP requests caching
+
+## Behavior
+
+We currently use Nginx to cache HTTP requests. It works like a standard reverse proxy: when a request comes in, a cache key is generated from various information sent by the client, a cached response is returned if one is found, otherwise the request is forwarded to the origin and returned to the client while being put in cache if appropriate.
+
+For the cache key, we take into account the following parameters:
+
+- Country corresponding to the client IP using GeoIP
+- Value of the `frontend_active_experiments` cookie (parsed from the `Cookie` request header) - its absence being a separate value as far as the cache key is concerned
+- The following HTTP headers:
+  - `Accept-Encoding`
+  - `User-Agent`
```

Comment:
Does the cache key include the entire `User-Agent` string? Stuart had mentioned hearing that doing that is essentially equivalent to not caching at all, due to the possible number of values. This is one of the reasons that I am curious about what our current cache hit ratio is. Are we trying to protect against a problem that already exists?

Reply:
It does include the entire string - that's done by the browser (we `Vary` on it). Ideally we would be smarter about it, but it's tricky since we need the browser name, version number and whether or not it's a mobile version for Firefox (we don't need as much for non-Firefoxes, though), and doing that would require adding logic in Nginx (that we would need to duplicate when switching to the CDN).

That certainly does have an impact on cache hit ratio, but our LCP numbers in the wild match WebPageTest numbers performed with cached responses. Ultimately we have a lot of traffic and sure, outliers on older versions of browsers are unlikely to get a cached response, but for the majority of users that should be fine.

Reply:
(I would also very much like to have stats about hit ratio)

Reply:
We think we can do something about that through CloudFront functions once on the CDN. There is a discussion about that in the middle of https://github.com/mozilla/addons-frontend/issues/10770 which I'll extract to a separate issue later.